Sources/HummingbirdHTTP2/HTTP2Channel.swift

```diff
@@ -0,0 +1,97 @@
+        get { http1.responder }
+        set { http1.responder = newValue }
+    }
+
+    public init(
+        tlsConfiguration: TLSConfiguration,
+        http1: HTTP1Channel = HTTP1Channel(),
+        responder: @escaping @Sendable (HBHTTPRequest, Channel) async throws -> HBHTTPResponse = { _, _ in throw HBHTTPError(.notImplemented) }
+    ) {
+        self.tlsConfiguration = tlsConfiguration
+        self.http1 = http1
+        // self.additionalChannelHandlers = additionalChannelHandlers
+        self.responder = responder
+    }
+
+    public func initialize(channel: Channel, configuration: HBServerConfiguration, logger: Logger) -> EventLoopFuture<Value> {
+        channel.eventLoop.flatSubmit {
+            do {
+                let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
+                try channel.pipeline.syncOperations.addHandler(NIOSSLServerHandler(context: sslContext))
+            } catch {
+                return channel.eventLoop.makeFailedFuture(error)
+            }
+            
+            return channel.configureAsyncHTTPServerPipeline { http1Channel -> EventLoopFuture<HTTP1Channel.Value> in
+                http1Channel
+                    .pipeline
+                    .addHandler(HBHTTPSendableResponseChannelHandler())
+                    .flatMapThrowing {
+                        try HTTP1Channel.Value(synchronouslyWrapping: http1Channel)
+                    }
+            } http2ConnectionInitializer: { http2Channel -> EventLoopFuture<NIOAsyncChannel<HTTP2Frame, HTTP2Frame>> in
+                http2Channel.eventLoop.makeCompletedFuture {
+                    try NIOAsyncChannel<HTTP2Frame, HTTP2Frame>(synchronouslyWrapping: http2Channel)
+                }
+            } http2StreamInitializer: { http2ChildChannel -> EventLoopFuture<HTTP1Channel.Value> in
+                http2ChildChannel
+                    .pipeline
+                    .addHandlers(HTTP2FramePayloadToHTTP1ServerCodec(), HBHTTPSendableResponseChannelHandler())
+                    .flatMapThrowing {
+                        try HTTP1Channel.Value(synchronouslyWrapping: http2ChildChannel)
+                    }
+            }
+        }
+    }
+
+    public func handle(value: Value, logger: Logger) async {
+        do {
+            let channel = try await value.get()
+            switch channel {
+            case .http1_1(let http1):
+                await handleHTTP(asyncChannel: http1, logger: logger)
+            case .http2((let http2, let multiplexer)):
+                try await withThrowingDiscardingTaskGroup { group in
+                    for try await client in multiplexer.inbound {
+                        group.addTask {
+                            await handleHTTP(asyncChannel: client, logger: logger)
+                        }
+                    }
+                }
+
+                try await http2.channel.close()
```

Comment:
The reason I close explicitly is because _not_ doing so would deinit `http2` implicitly closing the connection. It's simply necessary for ensuring Swift retains that Channel, nothing more.

Reply:
The reason I close explicitly is because _not_ doing so would deinit `http2` implicitly closing the connection. It's simply necessary for ensuring Swift retains that Channel, nothing more. But we're really not interested in reading/writing to the channel